dali/operators/input/video_input.cc

```diff
@@ -40,23 +40,33 @@ VideoInput<Backend, FramesDecoder>::SetNextDataIdTrace(Workspace &ws, std::strin
 }
 
 
-template<>
-bool VideoInput<CPUBackend>::SetupImpl(std::vector<OutputDesc> &output_desc,
-                                       const Workspace &ws) {
+template<typename Backend, typename FramesDecoder>
+void VideoInput<Backend, FramesDecoder>::CreateDecoder(const Workspace &ws) {
+  auto sample = encoded_video_[0];
+  auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
+  size_t size = sample.shape().num_elements();
+  if constexpr (is_cpu) {
```

Comment:
What's the point of Having FramesDecoder as a template parameter to later bypass it and use FramesDecoderGpu explicitly? Can we rely on `FramesDecoder` in both cases?

Reply:
Good point. I'll change to `FramedDecoder`